File: src/main/java/pl/edu/pbs/request/spp/SppRequest.java
package pl.edu.pbs.request.spp;

import lombok.Data;
import pl.edu.pbs.spp.Edge;
import pl.edu.pbs.spp.Node;
import pl.edu.pbs.spp.PathRequirement;
import pl.edu.pbs.spp.RoadType;

import java.util.List;

@Data
public class SppRequest {

    private List<Node> nodes;
    private List<RoadType> roadTypes;
    private List<Edge> edges;
    private PathRequirement pathRequirement;

}

File: tsp-solver-frontend/src/app/model/request/spp/SppRequest.ts
import {NodeRequest} from "./NodeRequest";
import {EdgeRequest} from "./EdgeRequest";
import {PathRequirement} from "./PathRequirement";
import {RoadTypeRequest} from "./RoadTypeRequest";

export class SppRequest {
  nodes: NodeRequest[];
  roadTypes: RoadTypeRequest[];
  edges: EdgeRequest[];
  pathRequirement: PathRequirement;

  constructor(nodes: NodeRequest[], roadTypes: RoadTypeRequest[], edges: EdgeRequest[], pathRequirement: PathRequirement) {
    this.nodes = nodes;
    this.roadTypes = roadTypes;
    this.edges = edges;
    this.pathRequirement = pathRequirement;
  }

}
